src/pages/index.js

```diff
@@ -0,0 +1,300 @@
+    if (openModal) {
+      closeModal(openModal);
+    }
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  modal.addEventListener("click", closeOverlay);
+  document.addEventListener("keydown", handleEsc);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  modal.removeEventListener("click", closeOverlay);
+  document.removeEventListener("keydown", handleEsc);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, "Save");
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, "Saving...");
+  const cardData = { name: cardNameInput.value, link: cardLinkInput.value };
+
+  api
+    .createCard(cardData)
+    .then((data) => {
+      const cardElement = getCardElement(data, { _id: "currentUserId" }); // Pass userInfo or mock it
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      evt.target.reset();
+    })
+    .catch((err) => {
+      console.error("Error creating card:", err);
+    })
+    .finally(() => {
+      setButtonText(submitBtn, "Save");
+    });
+}
+
+function handleDeleteCardSubmit(cardElement, cardId) {
+  openModal(deleteModal);
+
+  // Select the confirmation button inside the delete modal
+  const deleteConfirmButton = deleteModal.querySelector(
+    ".modal__confirm-button"
+  );
+
+  // Add a one-time event listener for the confirmation
+  deleteConfirmButton.addEventListener(
+    "click",
+    () => {
+      // Set the button text to "Deleting..."
+      setButtonText(deleteConfirmButton, (defaultText = "Deleting..."));
+
+      api
+        .deleteCard(cardId)
+        .then(() => {
+          cardElement.remove(); // Remove the card from the DOM
+          closeModal(deleteModal); // Close the modal
+        })
+        .catch((err) => {
+          console.error("Error deleting card:", err);
+        })
+        .finally(() => {
+          // Reset the button text to "Yes" (or the default text)
+          setButtonText(deleteConfirmButton, (loadingText = "Yes"));
+        });
+    },
+    { once: true } // Ensure the event listener is only triggered once
+  );
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, "Saving...");
+  const avatarUrl = avatarInput.value;
+
+  api
+    .updateAvatar(avatarUrl)
+    .then((data) => {
+      const profileAvatar = document.querySelector(".profile__avatar");
```

Comment:
`document.querySelector(".profile__avatar")`

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine